cli/azd/pkg/tools/azcli/spring_app.go

```diff
@@ -0,0 +1,250 @@
+	userAgent          string
+}
+
+// Creates a new instance of the NewSpringService
+func NewSpringService(
+	credentialProvider account.SubscriptionCredentialProvider,
+	httpClient httputil.HttpClient,
+) SpringService {
+	return &springService{
+		credentialProvider: credentialProvider,
+		httpClient:         httpClient,
+		userAgent:          azdinternal.MakeUserAgentString(""),
+	}
+}
+
+type SpringAppProperties struct {
+	Fqdn []string
+}
+
+func (ss *springService) GetSpringAppProperties(
+	ctx context.Context,
+	subscriptionId, resourceGroup, instanceName, appName string,
+) (*SpringAppProperties, error) {
+	client, err := ss.createSpringAppClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	springApp, err := client.Get(ctx, resourceGroup, instanceName, appName, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed retrieving spring app properties: %w", err)
+	}
+
+	return &SpringAppProperties{
+		Fqdn: []string{*springApp.Properties.Fqdn},
+	}, nil
+}
+
+func (ss *springService) UploadSpringArtifact(
+	ctx context.Context,
+	subscriptionId, resourceGroup, instanceName, appName, artifactPath string,
+) (*string, error) {
+	credential, err := ss.credentialProvider.CredentialForSubscription(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	options := clientOptionsBuilder(ss.httpClient, ss.userAgent).BuildArmClientOptions()
+
+	file, err := os.Open(artifactPath)
+	if err != nil {
+		return nil, fmt.Errorf("artifact do not exists: %s", artifactPath)
+	}
+	defer file.Close()
+
+	springClient, err := armappplatform.NewAppsClient(subscriptionId, credential, options)
+	if err != nil {
+		return nil, fmt.Errorf("creating SpringApps client: %w", err)
+	}
+	storageInfo, err := springClient.GetResourceUploadURL(ctx, resourceGroup, instanceName, appName, nil)
+	if err != nil {
+		return nil, fmt.Errorf("get resource upload URL failed: %w", err)
+	}
+
+	url, _ := url.Parse(*storageInfo.UploadURL)
+
+	fileURL := azfile.NewFileURL(*url, azfile.NewPipeline(azfile.NewAnonymousCredential(), azfile.PipelineOptions{}))
+	err = azfile.UploadFileToAzureFile(ctx, file, fileURL,
+		azfile.UploadToAzureFileOptions{
+			Parallelism: 3,
+			Metadata: azfile.Metadata{
+				"createdby": "AZD",
+			},
+		})
+
+	if err != nil {
+		return nil, fmt.Errorf("artifact '%s' upload failed: %w", artifactPath, err)
+	}
+
+	return storageInfo.RelativePath, nil
+}
+
+func (ss *springService) DeploySpringAppArtifact(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroup string,
+	instanceName string,
+	appName string,
+	relativePath string,
+	deploymentName string,
+) (*string, error) {
+	deploymentClient, err := ss.createSpringAppDeploymentClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	springClient, err := ss.createSpringAppClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	poller, err := deploymentClient.BeginCreateOrUpdate(ctx, resourceGroup, instanceName, appName, deploymentName,
```

Comment:
Would have expected that we would want to wait for the poller returned by the `BeginCreateOrUpdate` call here to complete before we try to set it as the active deployment?  Are we opening ourselves up to races here were the LRO is in flight when we try to set it as the active deployment?

Additionally, I would have expected that we would also wait for the poller returned by `BeginSetActiveDeployments` to complete before we said this whole operation is complete.

If running these two operations concurrently and also ignoring the result of the `BeginSetActiveDeployments` is correct, could we add a comment to this effect around L160 to explain why it's safe to do both of these things?